api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1340,6 +1410,9 @@ private HttpResponse executePut(String bucketName, String objectName, Map<String
     throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
            InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
            InternalException {
+    if (headerMap != null) {
+      headerMap = amzPrefixHeaderMap(headerMap);
```

Comment:
This should be done in `execute()`

Reply:
But this change will validate GET requests also right ?

Reply:
Done